02/app.js

```diff
@@ -1 +1,11 @@
-console.log('DOM');
\ No newline at end of file
+console.log('DOM');
+
+const linksEl = document.querySelectorAll('[data-url]');
+linksEl.forEach((link) => {
+  const url = link.dataset.url;
+  if (url && url !== 'null') {
```

Comment:
To sprawdzenie nie jest potrzebne - szykujesz elementy tylko z odpowiednim atrybutem tj `data-url` więc jak on nie będzie istniał to element nie zostanie wyszukany.

Zamiast linii od 6 do 9, wystarczy sama linia nr 7